src/integrationTest/java/org/opensearch/security/rest/WhoAmITests.java

```diff
@@ -169,4 +153,187 @@ public void testWhoAmIPost() {
         }
 
     }
+
+    @Test
+    public void testAuditLogSimilarityWithTransportLayer() {
+        try (TestRestClient client = cluster.getRestClient(AUDIT_LOG_VERIFIER)) {
+            assertResponse(client.get(WHOAMI_PROTECTED_ENDPOINT), HttpStatus.SC_OK, expectedAuthorizedBody);
+            assertExactlyOnePrivilegeEvaluationMessage(GRANTED_PRIVILEGES, AUDIT_LOG_VERIFIER);
+
+            assertThat(client.get("_cat/indices").getStatusCode(), equalTo(HttpStatus.SC_OK));
+
+            // transport layer audit messages
+            auditLogsRule.assertExactly(2, grantedPrivilege(AUDIT_LOG_VERIFIER, "GetSettingsRequest"));
+
+            List<AuditMessage> grantedPrivilegesMessages = auditLogsRule.getCurrentTestAuditMessages()
+                .stream()
+                .filter(msg -> msg.getCategory().equals(GRANTED_PRIVILEGES))
+                .collect(Collectors.toList());
+
+            verifyAuditLogSimilarity(grantedPrivilegesMessages);
+        }
+    }
+
+    private void assertResponse(TestRestClient.HttpResponse response, int expectedStatus, String expectedBody) {
+        assertThat(response.getStatusCode(), equalTo(expectedStatus));
+        assertThat(response.getBody(), equalTo(expectedBody));
+    }
+
+    private void assertExactlyOneAuthenticatedLogMessage(TestSecurityConfig.User user) {
+        auditLogsRule.assertExactly(
+            1,
+            userAuthenticated(user).withLayer(AuditLog.Origin.REST)
+                .withRestMethod(RestRequest.Method.GET)
+                .withRequestPath("/" + WhoAmITests.WHOAMI_PROTECTED_ENDPOINT)
+                .withInitiatingUser(user)
+        );
+    }
+
+    private void assertExactlyOnePrivilegeEvaluationMessage(AuditCategory privileges, TestSecurityConfig.User user) {
+        auditLogsRule.assertExactly(
+            1,
+            auditPredicate(privileges).withLayer(AuditLog.Origin.REST)
+                .withRestMethod(RestRequest.Method.GET)
+                .withRequestPath("/" + WhoAmITests.WHOAMI_PROTECTED_ENDPOINT)
+                .withEffectiveUser(user)
+        );
+    }
+
+    private void verifyAuditLogSimilarity(List<AuditMessage> currentTestAuditMessages) {
+        List<AuditMessage> restSet = new ArrayList<>();
+        List<AuditMessage> transportSet = new ArrayList<>();
+
+        // It is okay to loop through all even though we end up using only 2, as the total number of messages should be around 8
+        for (AuditMessage auditMessage : currentTestAuditMessages) {
```

Comment:
Just trynna confirm: the `currentTestAuditMessages` should include both `rest` and `transport` sets right? And the total number is just around 8?

Reply:
yes, and yes the number is 8 for these tests.